bin/oio-check-directory.py

```diff
@@ -0,0 +1,212 @@
+                    assert registered == known
+                except Exception as ex:
+                    logging.exception(
+                            "Failed to list the M0 from the ZK: %s", ex)
+                finally:
+                    zh.close()
+
+    # Check the state of the META1 services
+    if 'meta1' in args.actions:
+        logging.debug("Checking the META1 services")
+
+        # All the services must have been declared
+        c0 = list(filter_services(catalog, 'meta1'))
+        l0 = list(filter_services(live, 'meta1'))
+        assert len(c0) == len(l0)
+        logging.info("All the META1 are alive")
+
+        # They also need a positive score
+        for _, _, _, m1_score in l0:
+            assert m1_score > 0
+        logging.info("All the META1 have a positive score")
+
+    # Check the directory has been bootstraped
+    if 'dir' in args.actions:
+        import subprocess
+        from oio.directory.meta0 import Meta0Client
+        from oio.common.utils import json
+
+        logging.debug("Checking the directory bootstrap")
+
+        # Get an official dump from the proxy, check its size
+        m0 = Meta0Client({"namespace": args.ns})
+        prefixes = m0.list()
+        assert len(prefixes) == 65536
+        logging.info("The proxy serves a full META0 dump")
+
+        # contact each M0 to perform a check: any "get" command will
+        # fail if the meta0 is not complete. Unfortunately we just have
+        # oio-meta0-client to target a specific service.
+        for t, i, p, s in filter_services(catalog, 'meta0'):
+            url = '%s:%d' % (i, p)
+            subprocess.check_call(['oio-meta0-client', url, 'get', '0000'])
+        logging.info("All the META0 are complete")
+
+        # contact each meta0 to check that all the dumps are identical
+        dump0 = None
+        for t, i, p, s in filter_services(catalog, 'meta0'):
+            url = '%s:%d' % (i, p)
+            dump = subprocess.check_output(['oio-meta0-client', url, 'list'])
+            if dump0 is None:
+                dump0 = dump
+            else:
+                assert dump0 == dump
+        logging.info("All the META0 are the same")
+
+        # Check all the meta1 are concerned
+        reverse_dump = set()
+        for _, v in json.loads(dump0).iteritems():
+            for url in v:
+                reverse_dump.add(url)
+        m1 = list(filter_services(catalog, 'meta1'))
+        assert len(m1) == len(reverse_dump)
+        logging.info("All the META1 have been assigned")
+
+    # Check the RDIR status
+    if 'rdir' in args.actions:
+        from oio.rdir.client import RdirDispatcher
+
+        logging.debug("Checking the RDIR services")
+
+        # Load the assigned rdir services
+        client = RdirDispatcher({"namespace": args.ns})
+        all_rawx, all_rdir = client.get_assignation()
+        all_rawx = [rawx for rawx in all_rawx if 'rdir' in rawx]
+        assert len(all_rawx) == len(all_rdir)
```

Comment:
It's not mandatory to have the same number of rdir and rawx services. And by the way, `get_assignation` already does the test. We may modify it to spit an exception instead of just logging a missing assignation (selectable with a boolean).